core/src/main/java/org/apache/calcite/sql/SqlDelete.java

```diff
@@ -32,7 +32,20 @@
  */
 public class SqlDelete extends SqlCall {
   public static final SqlSpecialOperator OPERATOR =
-      new SqlSpecialOperator("DELETE", SqlKind.DELETE);
+      new SqlSpecialOperator("DELETE", SqlKind.DELETE) {
+        @SuppressWarnings("argument.type.incompatible")
+        @Override public SqlCall createCall(
+            @Nullable SqlLiteral functionQualifier,
+            SqlParserPos pos,
+            @Nullable SqlNode... operands) {
+          return new SqlDelete(
+              pos,
+              operands[0],
+              operands[1],
+              operands.length > 2 ? (SqlSelect) operands[2] : null,
```

Comment:
Is accessing operands[3] and operands[4] safe here? The creation should receive exactly 4 parameters, right?

Reply:
To be exact, we don't know where the operands are taken from. The only thing we can be sure of is that there are exactly 4 parameters when getting them from the getOperandList method. In other cases, we cannot be sure of the number of operands. @xiedeyantu

Reply:
I looked at the createCall method of other sqlcall operators, and it seems that none of them check the length. I keep it the same?😁

Reply:
Yes, I think you can keep same style.